src/mstore.erl

```diff
@@ -390,14 +391,6 @@ do_put(MStore = #mstore{size=S, data_size = DataSize}, Metric,
     {ok, F1} = write(F, DataSize, Metric, Time, Data),
     do_put(MStore, Metric, R, DataRest, [{FileBase, F1} | FileRest]);
 
-do_put(MStore = #mstore{size=S, data_size = DataSize}, Metric,
-       [{Time, Size} | R], InData,
-       [First, {FileBase, F}]) when
-      ((Time div S)*S) =:= FileBase ->
```

Comment:
there is a when condition here that does make the 1st and second condition differ